src/org/ods/services/BitbucketService.groovy

```diff
@@ -309,6 +309,50 @@ class BitbucketService {
         logger.debugClocked("buildstatus-${buildName}-${state}")
     }
 
+    /**
+     * Creates a code insight report in bitbucket via API.
+     * For further information visit https://developer.atlassian.com/server/bitbucket/how-tos/code-insights/
+     *
+     * @param result One of: PASS, FAIL
+     */
+    @SuppressWarnings('ParameterCount')
+    void createCodeInsightReport(String link, String repo, String gitCommit, String title, String details,
+                                 String result) {
+        withTokenCredentials { username, token ->
+            def payload = "{" +
+                "\"title\":\"${title}\"," +
+                "\"reporter\":\"OpenDevStack\"," +
+                "\"createdDate\":${System.currentTimeMillis()}," +
+                "\"details\":\"${details}\"," +
+                "\"result\":\"${result}\"," +
+                "\"data\": [" +
+                "{" +
+                "\"title\":\"Link\"," +
+                "\"value\":{\"linktext\":\"${link}\",\"href\":\"${link}\"}," +
+                "\"type\":\"LINK\"" +
+                "}" +
+                "]" +
+                "}"
+            try {
+                script.sh(
+                    label: 'Create Bitbucket Code Insight report via API',
+                    script: """curl \\
+                        --fail \\
+                        -sS \\
+                        --request PUT \\
+                        --header \"Authorization: Bearer ${token}\" \\
+                        --header \"Content-Type: application/json\" \\
+                        --data '${payload}' \\
+                        ${bitbucketUrl}/rest/insights/latest/projects/${project}/\
```

Comment:
Is `latest` the right value? The other API calls in this service target `1.0`?

Reply:
Yes, the url in the doc https://docs.atlassian.com/bitbucket-server/rest/7.11.2/bitbucket-code-insights-rest.html#idm45774552049392 did not work, it took me half a day to find the working url in some PR of another project on github that integrated that. super strange.

Reply:
Oh Bitbucket.

Maybe worth a comment in the source file to document why it is like this ...

Reply:
I just checked again, they fixed it meanwhile, I adjusted accordingly.

Reply:
@renedupont are you able to list which Bitbucket version the URL works with and with which version it doesn't? We use a specific version in the corporate context which might differ from the one used at the BI-X.

Reply:
I think the initial confusion I had was that they have a wrong url in this how-to https://developer.atlassian.com/server/bitbucket/how-tos/code-insights/#create but not in the API doc. I changed it now from latest to 1.0 and I also compared for our bitbucket versions (which differ only in patch versions) and the url did not change.

Reply:
@metmajer As I understand Atlassian versioning, v1.0 means Bitbucket Server/DataCenter, and v2.0 means Bitbucket Cloud. All of shared lib targets v1.0 of the API.